### PR TITLE
Fix 'Receive fact-check' action button is displayed as 'Unsupported message in the conversation history

### PR DESCRIPTION
**Pull request by danielevalverde**
## Description

- add the parser for the payload for a button message to get the text from the capi structure
- add a unit test for the Chat History component 

Reference: CV2-4032

## Type of change

- [ ] Performance improvement and/or refactoring (non-breaking change that keeps existing functionality)
- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Security mitigation or enhancement
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Automated test (add or update automated tests)

## Checklist

- [X] I have performed a self-review of my own code
- [ ] I've made sure my branch is runnable and given good testing steps in the PR description
- [ ] I considered secure coding practices when writing this code. Any security concerns are noted above.
- [X] I have commented my code in hard-to-understand areas, if any
- [ ] I have made needed changes to the README
- [ ] My changes generate no new warnings
- [X] I have added tests that prove my fix is effective or that my feature works
- [ ] If I implemented any new components, they are self-contained, their `propTypes` are declared and they use React Hooks and, if data-fetching is required, they use Relay Modern with fragment containers
- [ ] If my components involve user interaction - specifically button, text fields, or other inputs - I have added a [BEM-like class name](https://meedan.atlassian.net/wiki/spaces/ENG/pages/1327628289/Naming+conventions+for+interactive+elements) to the element that is interacted with
- [ ] To the best of my knowledge, any new styles are applied according to the design system
- [ ] If I added a new external dependency, I included a rationale for doing so and an estimate of the change in bundle size (e.g., checked in https://bundlephobia.com/)

### CODE REVIEW

File: src/app/components/layout/ChatHistory.test.js [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+          received: 1700674230,
+          quotedMessage: { content: { _id: '123343434ZDN' } },
+        },
+      ],
+    },
+    team_id: 1797,
+    created_at: new Date('Wed, 22 Nov 2023 17:30:34.652769000 UTC +00:00'),
+    updated_at: new Date('Wed, 22 Nov 2023 17:30:34.652769000 UTC +00:00'),
+    state: 'received',
+  };
```
Comment (amoedoamorim) [inline src/app/components/layout/ChatHistory.test.js:76]: Are these ids from actual db data? Can we strip them from the tests?

Comment (danielevalverde) [reply src/app/components/layout/ChatHistory.test.js:76]: > Are these ids from actual db data? Can we strip them from the tests?

All the ids and numbers are fake data for the test :)

Comment (amoedoamorim) [reply src/app/components/layout/ChatHistory.test.js:76]: Awesome

**Review by amoedoamorim — APPROVED** [review]
Looks good to me!

**Review by caiosba — APPROVED** [review]
Tested locally and works great!

Captura de tela de 2023-12-07 15-32-04

**Pull request merged**